addons/website_payment/controllers/main.py

```diff
@@ -22,7 +22,7 @@ def payment_method(self):
     def delete(self, delete_pm_id=None):
         if delete_pm_id:
             pay_meth = request.env['payment.method'].browse(int(delete_pm_id))
-            pay_meth.unlink()
+            pay_meth.sudo().unlink()
```

Comment:
do not unlink using sudo user.

give proper access rules for portal user to delete only own payment methods.